testsuite/tests/basic-more/safer_matching.ml

```diff
@@ -0,0 +1,11 @@
+(* TEST
+   flags += "-safer-matching"
+*)
+
+type 'a dyn = Int : int -> int dyn | Float : float -> float dyn;;
+
+let f (Float x) = x;;
+
+Format.printf "%f\n%!" (f (Float 3.1415));;
+
+try Printexc.print f (Obj.magic (Int 3)) with _ -> 0.;;
```

Comment:
This testcase would be much easier to understand with:
1. a typing annotation on f (`float dyn -> float`)
2. a typing annotation on the `Obj.magic` return type

(annotating `Obj.magic` is a good practice anyway to ensure that we know what type we mean.)